service/round.go

```diff
@@ -121,65 +96,44 @@ func (r *round) submit(key, challenge []byte) error {
 	return r.challengesDb.Put(key, challenge, &opt.WriteOptions{Sync: true})
 }
 
-func (r *round) numChallenges() int {
-	iter := r.challengesDb.NewIterator(nil, nil)
-	defer iter.Release()
-
-	var num int
-	for iter.Next() {
-		num++
-	}
-
-	return num
-}
-
-func (r *round) isEmpty() bool {
-	iter := r.challengesDb.NewIterator(nil, nil)
-	defer iter.Release()
-	return !iter.Next()
-}
-
 func (r *round) execute(ctx context.Context, end time.Time, minMemoryLayer uint) error {
 	logger := logging.FromContext(ctx).WithFields(log.String("round", r.ID))
 	logger.Info("executing until %v...", end)
 
-	r.executionStarted = time.Now()
+	r.ExecutionStarted = time.Now()
```

Comment:
Should `executionStarted` be private? Otherwise anyone with access to round can modify that field.

Reply:
I'd like it to be private too, but then serialization doesn't work (which is required to persist round's state). Do you know how to get around it?

Reply:
See [here](https://github.com/spacemeshos/poet/pull/195#discussion_r1054263010) and [here](https://github.com/spacemeshos/poet/pull/195#discussion_r1054263737) 😉